pkg/io/equal_binary.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io
+
+import (
+	"bytes"
+	"encoding"
+
+	"github.com/pkg/errors"
+)
+
+// EqualBinary returns whether the binary representation of the two values `a` and `b` are equal.
+// It returns an error when marshalling fails.
+func EqualBinary(a, b encoding.BinaryMarshaler) (bool, error) {
```

Comment:
Still needed when we have `Asset.Equal`?

Reply:
Yes, because when validating a state we use it to compare `Data`.

However, in a separate PR, we could implement `Data.Equal` and then remove this function.

Reply:
Created #280